### ISSUE TITLE
Investigate Possibility of Getting Map "Distance" Key

### ISSUE THREAD

**Issue by JRegimbal** [Shared-Reality-Lab/IMAGE-server#495]
The proposed distance key to the request (see #491) would be a good improvement, but would require adjustments to the request schema and to the extension. Before it can be added, we need to have a path to actually implementing this from the information available via the embedded google maps (or OSM, or another service). This issue will be resolved when there is a reasonable path to setting "distance" from the client, or an alternative that could be used in a similar way.

**Comment by samueljohnsegun148** [comment]
Okay. Adjustments to the request schema will definitely be necessary, so much more the extension as rightly said. Can I have some preliminary information/hints, maybe from autour, so as to have a few ideas on the extension? Maybe that would assist me in the path setting for the distance field.

**Comment by JRegimbal** [comment]
As far as I know, autour doesn't extract any information about distance from the graph (@jeffbl correct me if I'm wrong). The only thing similar that I've come across is the Google Maps zoom level (mentioned in [here](https://developers.google.com/maps/documentation/embed/embedding-map)) which does not explicitly indicate a scale. A google employee shared [a way of getting meters per pixel from zoom level](https://groups.google.com/g/google-maps-js-api-v3/c/hDRO4oHVSeM/m/osOYQYXg2oUJ) which may be reliable enough to use to estimate "distance" as proposed.

**Comment by jeffbl** [comment]
@JRegimbal No, I don't recall any scale information from Autour (but it has been a long time!) just positions of POIs and intersections, for rendering by Autour in an egocentric way. The goal here is to get the overal scale of the embedded map? Nothing like that in Autour if so, since Autour starts with just a lat/lng as reported by the phone GPS. Let me know if more detail is needed, or if I'm not answering the real question.

**Comment by samueljohnsegun148** [comment]
> As far as I know, autour doesn't extract any information about distance from the graph (@jeffbl correct me if I'm wrong). The only thing similar that I've come across is the Google Maps zoom level (mentioned in [here](https://developers.google.com/maps/documentation/embed/embedding-map)) which does not explicitly indicate a scale. A google employee shared [a way of getting meters per pixel from zoom level](https://groups.google.com/g/google-maps-js-api-v3/c/hDRO4oHVSeM/m/osOYQYXg2oUJ) which may be reliable enough to use to estimate "distance" as proposed.

Your information is very helpful to me. I think I should be able to have the distance parameter estimated, by adjusting the zoom level using the information provided with this link -https://wiki.openstreetmap.org/wiki/Slippy_map_tilenames#Resolution_and_Scale.

**Comment by JRegimbal** [comment]
Note that we currently don't support a embedded OSM maps, so there isn't a usable OSM zoom to get.

**Comment by samueljohnsegun148** [comment]
I know that modifying the request schema shouldn't be a big issue (I stand to be corrected). However, if I understand correctly, creating a path for the client to input the distance parameter is the major issue here, which I believe could be addressed from the extension side. Can I have some insights into how the extension works, in case it is something I can work on? This may also be a silly question, but can't we create a field for the "distance" on the extension side, just as there is for coordinates (lat/lon)?

**Comment by jeffbl** [comment]
Based on discussion at tech-arch last week, it seems that the other major issue is actually obtaining the scale of the map translated into an actual distance in terms of the scale. From the discussion above, it sounds like this may be difficult or impossible for Google maps, but would be easier for OSM embedded maps. When we are deciding what other maps formats to support, we should consider whether or not getting the overall scale is feasible, as part of the decision, especially if this informaiton allows us to create materially better renderings.

**Comment by jeffbl** [comment]
@samueljohnsegun148 I believe we're trying to complete other work items without the distance key work, so I believe this should be postponed to November sprint, or even to backlog if we don't yet have a viable path to obtaining distance information?

**Comment by jeffbl** [comment]
Moving to Nov. sprint. Will move to backlog if no viable path to getting this information.

**Comment by jeffbl** [comment]
Postponing to Dec19 sprint, but @samueljohnsegun148 if we don't have a viable path to getting this information, please indicate that here, and I'll move this to backlog since it will require more in-depth investigation.

**Comment by samueljohnsegun148** [comment]
I'm still doing some personal background work on it, but it should be moved to the backlog, since I think this can always be brought up later, when there is a significant progress/findings.
